### DISCUSSION TITLE
Local Port Issues

### DISCUSSION THREAD

**Discussion by Tokenomacs** [MetaMask/snaps#1127] [Q&A]
Hi Snap community,

I have been loving Snap development, and I have worked on two different snap projects thus far. I was using the template-snap interface for both Snaps each served at localhost port 8000 (let's call them projects 1 & 2). 

Recently, I deleted one of the snap project 1, and hoped the other snap project 2 would take its place and serve on local host port 8000.

However, whenever I connect my wallet and attempt to send a transaction, it still serves me to the functionality of the deleted Snap project 1.

So far, I have tried the following:
- deleting authorization/connection to the local host port 8000 within my Metamask wallet
- using npx kill-port to kill any remaining running nodes on local host port 8000 / 8080
- clearing my cache and browser history
- using my command line to taskkill

However, nothing has worked and I would like to test snap project 2, but have no way of doing so because snap project 1 seems to override its connectivity on local host port 8000.

Any suggestions?

**Comment by ritave** [comment]
Hello @Tokenomacs, I'm assuming you're not serving two snaps at the same time from the CLI.

We detect distinct snaps using their IDs, in this case it would be a snap with ID `local:http://localhost:8000`, since both snaps used the same ID, the extension might be detecting those two snaps as the same one.

What you could try doing is either one of:
* Removing the snap from the extension's list of installed snaps.
* Reinstalling the extension
* Using a different port for each snap, for example `8000` and `8001`.

**Reply by Tokenomacs** [reply]
Thanks for the help.

I tried bullet points 1 and 2 but it seems to be detecting both snaps as at 8000 port. 

Is there any simple way to serve the snap on a different port?

**Reply by ritave** [reply]
Assuming you're using the template-snap-monorepo, there are two places you need to update:
1. [The `snap.config.js`](https://github.com/MetaMask/template-snap-monorepo/blob/d9b6308ba69f59ab4c23c7a21b2b827fe982b304/packages/snap/snap.config.js#L4)
   * This configures the port on which the snap is being served from in the development environment
2. [The `config/snap.ts`](https://github.com/MetaMask/template-snap-monorepo/blob/main/packages/site/src/config/snap.ts#L6)
   * This configures on which port the website is trying to fetch the snap from

**Reply by Tokenomacs** [reply]
Thank you so much Ritave! You are the man.

**Comment by Tokenomacs** [comment]
![image](https://user-images.githubusercontent.com/92122220/215258252-90befeef-4e72-4c3d-a502-c147c7f8ed99.png)
Hi Ritave, unfortunately this still doesn't seem to be working. 

What I have tried thus far:
1. I reconfigured the snap.config.js and config/snap.ts files of the template-snap-monorepo to serve from different ports from which my first snap was developed on
2. I created a brand new flask wallet and cleared my cache
3. I used my OS command prompt to kill any threads hosted on the 8000 port (the frontend application in the picture), the port from which the snap is being served from, and the website port.

Is there anything else I could try?

**Reply by ritave** [reply]
Hey @Tokenomacs,

Could you create an issue out of this please and someone from our team will try to debug with you?

Will be helpful for that:
1. An open-source git repository with your current state of the problem
2. Any error logs from the Dapp developer console, the background.html developer console (chrome -> extensions (chrome://extensions) -> enable Developer Mode -> click background.html in the Flask extension)
3. Any logs from the CLI
